lib/collection/src/operations/conversions.rs

```diff
@@ -132,6 +138,165 @@ impl From<Record> for api::grpc::qdrant::RetrievedPoint {
+        match collection_info_response.result {
+            None => Err(Status::invalid_argument("Malformed CollectionInfo type")),
+            Some(collection_info_response) => {
+                Ok(Self {
+                    status: match collection_info_response.status {
+                        1 => CollectionStatus::Green,
+                        2 => CollectionStatus::Yellow,
+                        3 => CollectionStatus::Red,
+                        _ => {
+                            return Err(Status::invalid_argument("Malformed CollectionStatus type"))
+                        }
+                    },
+                    optimizer_status: match collection_info_response.optimizer_status {
+                        None => {
+                            return Err(Status::invalid_argument("Malformed OptimizerStatus type"))
+                        }
+                        Some(api::grpc::qdrant::OptimizerStatus { ok, error }) => {
+                            if ok {
+                                OptimizersStatus::Ok
+                            } else {
+                                OptimizersStatus::Error(error)
+                            }
+                        }
+                    },
+                    vectors_count: collection_info_response.vectors_count as usize,
+                    segments_count: collection_info_response.segments_count as usize,
+                    disk_data_size: collection_info_response.disk_data_size as usize,
+                    ram_data_size: collection_info_response.ram_data_size as usize,
+                    config: match collection_info_response.config {
+                        None => {
+                            return Err(Status::invalid_argument("Malformed CollectionConfig type"))
+                        }
+                        Some(config) => CollectionConfig {
+                            params: match config.params {
+                                None => {
+                                    return Err(Status::invalid_argument(
+                                        "Malformed CollectionParams type",
+                                    ))
+                                }
+                                Some(params) => CollectionParams {
+                                    vector_size: params.vector_size as usize,
+                                    distance: match params.distance {
+                                        1 => segment::types::Distance::Cosine,
+                                        2 => segment::types::Distance::Euclid,
+                                        3 => segment::types::Distance::Dot,
+                                        _ => {
+                                            return Err(Status::invalid_argument(
+                                                "Malformed CollectionParams distance",
+                                            ))
+                                        }
```

Comment:
Same here

Reply:
You can also specify it as an explicit discriminator - https://doc.rust-lang.org/rust-by-example/custom_types/enum/c_like.html

Reply:
done in https://github.com/qdrant/qdrant/pull/505/commits/5d903e6f909ef3550e1d2cca77cf55a5708ef50a